rewrite MIP-8 for clarity (#31)
monad-crypto/MIPs: MIPS/MIP-8.md

    return root
```

### Leaves of the Merkle Patricia Trie

The Merkle Patricia Trie commits to `{page_index_i: page_commit(page_i)}` pairs, where `page_commit(page_i)` is a 32-byte commitment to the contents of page `i`.

This trie incorporates the following modifications:

1. **Hash function**: BLAKE3.
2. **Leaf values**: for each `page_index`, the corresponding leaf value is the page commitment of the page at that index.
3. **Leaf placement**: each `page_index` uniquely determines a path from the MPT root to its leaf. This path is computed exactly as in a standard MPT, using the `page_index` as the key.
4. **Trie structure**: the MPT structure is otherwise unchanged; branch, extension, and leaf nodes follow the standard MPT rules.
5. **On-demand computation**: the value of each storage leaf is exactly 32 bytes, so `page_commit(page)` can be recomputed from the page contents whenever needed. No additional storage layout changes are required.
6. **Merkle proofs**: Merkle proofs for page commitments are unchanged from a standard MPT. However, such a proof only establishes that a particular page has been committed.

### Word Inclusion Proofs

> The page commitment can be thought of as a fixed-size Merkle tree stored inside the MPT leaf. Given a Merkle proof for a page commitment, we can prove the inclusion of any specific word within that page. To construct an inclusion proof for a particular word, we need the following:

Reviewer: ```suggestion
The page commitment can be thought of as a fixed binary Merkle tree with its root stored inside the MPT leaf. The inclusion proof for a particular word within the page commitment is:
```

Otherwise this reads to me a bit like same thing repeated

| Field            | Size     | Description                                                                                         |
| ---------------- | -------- | --------------------------------------------------------------------------------------------------- |
| `word_idx`       | usize    | Position of the word within the page (0–127). Caller-provided, not part of the cryptographic proof. |
| `word`           | 32 B     | Value of the target word.                                                                           |
| `sibling_word`   | 32 B     | Companion word in the same pair-leaf.                                                               |

> | `parent_sibling` | 6 × 32 B | Sibling hashes at each parent level (pair-leaf → root).                                             |

Reviewer: > pair-leaf → root

I'd say this is still ambiguous. I'd either drop it or expand to more verbose `6 levels from pair-leaf parent to subroot`, I have slight preference to drop.
